### PR TITLE
Use proxied auth endpoint for Cognito

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Close https://github.com/enso-org/cloud-v2/issues/1552
  - Switch to proxied endpoint for Cognito to enforce account lockout rate limits

### Important Notes
None

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] ~~Unit tests have been written where possible.~~
- [x] ~~If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.~~

### CODE REVIEW

**Comment by somebody1234** [comment]
self testing:
- login works
- registration works
- change password does not work (lambda panic)
- invalid login work (the network request fails, but it it is a lambda panic so it does not return a Cognito JSON error so Amplify crashes)

**Comment by somebody1234** [comment]
@jdunkerley converting this one to draft as (afaik) this one is *potentially* blocked on some backend bugs.

@indiv0 not sure if they have been fixed since and we are simply waiting on a re-deploy - the specific issue seems to be that the lambda is panicking on invalid logins, and we would really prefer if the lambda proxies back the error response given by Cognito instead.

File: app/gui/env.d.ts [hunk 1]
```diff
@@ -216,6 +216,8 @@ declare global {
       // @ts-expect-error The index signature is intentional to disallow unknown env vars.
       readonly ENSO_CLOUD_STRIPE_KEY?: string
       // @ts-expect-error The index signature is intentional to disallow unknown env vars.
+      readonly ENSO_CLOUD_AUTH_ENDPOINT: string
```
Comment (MrFlashAccount) [inline app/gui/env.d.ts:221]: We need to add this var into .env.example

**Review by MrFlashAccount — APPROVED** [review]

**Comment by PabloBuchu** [comment]
Looks like refresh token session is still being send directly to cognito? 

Screenshot 2024-11-18 at 11 23 10
Screenshot 2024-11-18 at 11 23 07

**Review by PabloBuchu — APPROVED** [review]

**Comment by PabloBuchu** [comment]
I fixed issues on the backend and will deploy shortly.

**Pull request merged**